src/bootstrap/flags.rs

```diff
@@ -247,20 +267,26 @@ To learn more about a subcommand, run `./x.py <subcommand> -h`",
                     "enable this to generate a Rustfix coverage file, which is saved in \
                         `/<build_base>/rustfix_missing_coverage.txt`",
                 );
+                run_stage(&mut opts)
             }
             "bench" => {
                 opts.optmulti("", "test-args", "extra arguments", "ARGS");
+                link_stage(&mut opts)
```

Comment:
It feels definitely wrong for bench and test to have different flags...

Reply:
The way I chose this is by looking at whether `src/rustc` was built for that stage. Or in other words, whether it does any work when you pass it `--stage 0`. So `build --stage 0 src/rustc` does work; `test --stage 0 src/test/ui` does nothing (it runs the beta compiler); `bench --stage 0 src/rustc` does work; etc.